File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/SFProgramModule.java
package shadow.pipeline;


public class SFProgramModule extends SFPipelineElement{

	protected SFProgramComponent[] components=new SFProgramComponent[0];
	
	public SFProgramModule() {
	}
	
	public SFProgramModule(String name) {
		setName(name);
	}

	public void setComponents(SFProgramComponent[] components) {
		this.components = components;
	}

	public SFProgramComponent[] getComponents() {
		return components;
	}
	
	@Override
	public SFProgramModule clone() {
		SFProgramModule module=new SFProgramModule(getName());
		module.components=components;
		return module;
	}
}

File: SFDataApplicationTestC++/shadow/pipeline/SFProgramModule.cpp
//
//  SFProgramModule.cpp
//  
//
//  Created by Alessandro Martinelli on 18/10/12.
//  Copyright (c) 2012 Alessandro Martinelli. All rights reserved.
//

#include "SFProgramModule.h"



namespace sf{
	SFProgramModule::SFProgramModule() {
	}

	SFProgramModule::~SFProgramModule(){
        for(unsigned int i=0;i<components.size();i++){
            delete components[i];
        }
    }

	SFProgramModule::SFProgramModule(string name) {
		setName(name);
	}

	void SFProgramModule::setComponents(vector<SFProgramComponent*> components) {
		this->components = components;
	}

	vector<SFProgramComponent*> SFProgramModule::getComponents() {
		return components;
	}

	SFProgramModule* SFProgramModule::clone() {
		SFProgramModule* module=new SFProgramModule(getName());
		module->components=components;
		return module;
	}
}
